aips/__init__.py

```diff
@@ -1,50 +1,56 @@
+from gc import collect
 import aips.environment as environment
-from aips.environment import SOLR_URL
 from engines.solr import SolrLTR, SolrSemanticKnowledgeGraph, SolrEntityExtractor, SolrSparseSemanticSearch
 from engines.solr.SolrEngine import SolrEngine
+from engines.solr.SolrCollection import SolrCollection
+
+from engines.opensearch.OpenSearchCollection import OpenSearchCollection
+from engines.opensearch.OpenSearchEngine import OpenSearchEngine
+from engines.opensearch.OpenSearchLTR import OpenSearchLTR
+from engines.opensearch.OpenSearchSparseSemanticSearch import OpenSearchSparseSemanticSearch
 
 import os
-from IPython.display import display,HTML
+from IPython.display import display, HTML
 import pandas
 import re
-import requests
 
-engine_type_map = {"SOLR": SolrEngine()}
+engine_type_map = {"SOLR": SolrEngine(),
+                   "OPENSEARCH": OpenSearchEngine()}
 
-def get_engine():
-    return engine_type_map[environment.get("AIPS_SEARCH_ENGINE", "SOLR")]
+def get_engine(override=None):
+    engine_name = override.upper() if override else environment.get("AIPS_SEARCH_ENGINE", "SOLR")
+    return engine_type_map[engine_name]
 
-def set_engine(engine_name):    
+def set_engine(engine_name):
     engine_name = engine_name.upper()
     if engine_name not in engine_type_map:
         raise ValueError(f"No search engine implementation found for {engine_name}")
     else:
         environment.set("AIPS_SEARCH_ENGINE", engine_name)
 
-def get_ltr_engine(collection):
-    return SolrLTR(collection)
+def get_ltr_engine(collection):    
+    ltr_engine_map = {SolrCollection: SolrLTR,
+                      OpenSearchCollection: OpenSearchLTR}
+    return ltr_engine_map[type(collection)](collection)
 
 def get_semantic_knowledge_graph(collection):
-    return SolrSemanticKnowledgeGraph(collection)
+    return SolrSemanticKnowledgeGraph(get_engine("solr").get_collection(collection.name))
 
 def get_entity_extractor(collection):
-    return SolrEntityExtractor(collection)
+    return SolrEntityExtractor(get_engine("solr").get_collection(collection.name))
 
 def get_sparse_semantic_search():
-    return SolrSparseSemanticSearch()
+    SSS_map = {SolrEngine: SolrSparseSemanticSearch,
+               OpenSearchEngine: OpenSearchSparseSemanticSearch}
+    return SSS_map[type(get_engine())]()
 
 def healthcheck():
     try:
         if get_engine().health_check():
-            print("Solr is up and responding.")
```

Comment:
I think we still need to include the specific services running. The manuscript (appendix) states:

> ...will result in the healthcheck executing to confirm that all Docker containers are running and that the services running within them are healthy and responding. 

It also includes a screenshot which shows Solr and Zookeeper "up and responding"..

Reply:
I needed to put this code back in the oven for a second, it wasn't quite done yet. It's fully baked now.